src/main/java/de/tum/cit/aet/artemis/programming/web/ProgrammingExerciseResource.java

```diff
@@ -547,6 +550,26 @@ public ResponseEntity<ProgrammingExercise> getProgrammingExerciseWithAuxiliaryRe
         return ResponseEntity.ok(programmingExercise);
     }
 
+    /**
+     * GET /programming-exercises : Queries a programming exercise by its project key.
+     *
+     *
+     * @param projectKey the project key of the programming exercise
+     *
+     * @return the ProgrammingExercise of this project key in an ResponseEntity or 404 Not Found if no exercise exists
+     */
+    @GetMapping("programming-exercises/project-key/{projectKey}")
+    @EnforceAtLeastStudent
+    @AllowedTools(ToolTokenType.SCORPIO)
+    public ResponseEntity<ProgrammingExercise> getExerciseByProjectKey(@PathVariable String projectKey) {
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+
+        final ProgrammingExercise exercise = programmingExerciseRepository.findWithStudentParticipationLatestResultFeedbackTestCasesByProjectKey(user.getId(), projectKey)
```

Comment:
Please verify that this exercise is already released, otherwise a student should NOT have access to. Additionally make sure to filter all sensitive information before return this to the client. 

It can e.g. be the case that the student should not be able to see the last result, in case it is manual feedback from a tutor, but the assessment due date has not passed yet.